internal/provider/provider.go

```diff
@@ -161,6 +161,7 @@ func (p *HumanitecProvider) Resources(ctx context.Context) []func() resource.Res
 		NewResourceDefinitionResource,
```

Comment:
Unrelated to this line, could you add an example similar to https://github.com/humanitec/terraform-provider-humanitec/tree/main/examples/resources/humanitec_agent and run `go generate ./...` to update the docs?